sdk/go/pulumi-language-go/language_test.go

```diff
@@ -0,0 +1,269 @@
+	if os.Getenv("PULUMI_LANGUAGE_TEST_SHOW_FULL_OUTPUT") != "true" {
+		// Cut down logs so they don't overwhelm the test output
+		if len(message) > 1024 {
+			message = message[:1024] + "... (truncated, run with PULUMI_LANGUAGE_TEST_SHOW_FULL_OUTPUT=true to see full logs))"
+		}
+	}
+
+	if e.previousMessage == message {
+		e.logRepeat++
+		return &pbempty.Empty{}, nil
+	}
+
+	if e.logRepeat > 1 {
+		e.t.Logf("Last message repeated %d times", e.logRepeat)
+	}
+	e.logRepeat = 1
+	e.previousMessage = message
+
+	if req.StreamId != 0 {
+		e.t.Logf("(%d) %s[%s]: %s", req.StreamId, sev, req.Urn, message)
+	} else {
+		e.t.Logf("%s[%s]: %s", sev, req.Urn, message)
+	}
+	return &pbempty.Empty{}, nil
+}
+
+func runEngine(t *testing.T) string {
+	// Run a gRPC server that implements the Pulumi engine RPC interface. But all we do is forward logs on to T.
+	engine := &hostEngine{t: t}
+	stop := make(chan bool)
+	t.Cleanup(func() {
+		close(stop)
+	})
+	handle, err := rpcutil.ServeWithOptions(rpcutil.ServeOptions{
+		Cancel: stop,
+		Init: func(srv *grpc.Server) error {
+			pulumirpc.RegisterEngineServer(srv, engine)
+			return nil
+		},
+		Options: rpcutil.OpenTracingServerInterceptorOptions(nil),
+	})
+	require.NoError(t, err)
+	return fmt.Sprintf("127.0.0.1:%v", handle.Port)
+}
+
+func runTestingHost(t *testing.T) (string, testingrpc.LanguageTestClient) {
+	// We can't just go run the pulumi-test-language package because of
+	// https://github.com/golang/go/issues/39172, so we build it to a temp file then run that.
+	binary := t.TempDir() + "/pulumi-test-language"
+	cmd := exec.Command("go", "build", "-C", "../../../cmd/pulumi-test-language", "-o", binary)
+	output, err := cmd.CombinedOutput()
+	t.Logf("build output: %s", output)
+	require.NoError(t, err)
+
+	cmd = exec.Command(binary)
+	stdout, err := cmd.StdoutPipe()
+	require.NoError(t, err)
+	stderr, err := cmd.StderrPipe()
+	require.NoError(t, err)
+	stderrReader := bufio.NewReader(stderr)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		for {
+			text, err := stderrReader.ReadString('\n')
+			if err != nil {
+				wg.Done()
+				return
+			}
+			t.Logf("engine: %s", text)
+		}
+	}()
+
+	err = cmd.Start()
+	require.NoError(t, err)
+
+	stdoutBytes, err := io.ReadAll(stdout)
+	require.NoError(t, err)
+
+	address := string(stdoutBytes)
+
+	conn, err := grpc.Dial(
+		address,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithUnaryInterceptor(rpcutil.OpenTracingClientInterceptor()),
+		grpc.WithStreamInterceptor(rpcutil.OpenTracingStreamClientInterceptor()),
+		rpcutil.GrpcChannelOptions(),
+	)
+	require.NoError(t, err)
+
+	client := testingrpc.NewLanguageTestClient(conn)
+
+	t.Cleanup(func() {
+		assert.NoError(t, cmd.Process.Kill())
+		wg.Wait()
+		// We expect this to error because we just killed it.
+		contract.IgnoreError(cmd.Wait())
+	})
+
+	engineAddress := runEngine(t)
+	return engineAddress, client
+}
+
+var expectedFailures = map[string]string{
```

Comment:
Is it worth adding a comment like this (either linking to https://github.com/pulumi/pulumi/issues/15205, or something else?)
```suggestion
// TODO: These tests are not working yet because of issues in programgen
var expectedFailures = map[string]string{
```